R/check_filename.R

```diff
@@ -1,4 +1,38 @@
-#' Check the style of file names
+#' Check the style of file and folder names
+#'
+#' A consistent naming schema avoids problems when working together.
+#' Especially when working with different OS.
+#' Some OS (e.g. Windows) are case-insensitive whereas others (e.g. Linux) are
+#' case-sensitive.
+#' Note that the the checklist GitHub Actions will test your code on Linux,
+#' Windows and MacOS.
+#'
+#' The sections below describe the default rules.
+#' We allow several exceptions when the community standard is different.
+#' E.g. a package stores the function scripts in the `R` folder, while our
+#' standard we enforces lower case folder names.
+#' Use the the community standard, even if it does conform with the checklist
+#' rules.
+#' Most likely checklist will have an exception for the name.
+#' If not, you can file an [issue](https://github.com/inbo/checklist/issues) and
+#' motivate why you think we should add an exception.
```

Comment:
To me it seems better to start with the rules and mention the exceptions after the rules.  I just read this part, but it seems rather vague as I haven't read the rules yet.  So for clarity, just mention these exceptions each time as an additional rule.  For instance, under Rules for folder names, Folder names should only contain lower case letters, numbers and underscore (`_`)., add a rule `Exception: community standard folders can/must have an upper case: 'R'`, similar as what you did under Rules for file names.  If it is impossible to cover all exceptions, use `e.g.`.

And I would put the request for additional exceptions after the rules, in this way it refers to all rules and exceptions, not only for the one that is mentioned in this first part.